### PR TITLE
[Feature] Support external catalog ddl

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Feature: Support external catalog ddl for dla connector, related to issue https://github.com/StarRocks/starrocks/issues/5762

## Problem Summary(Required) ：

This pr is to support ddl like `create external catalog catalog_name properties ("type"="hive", "hive.metastore.uris"="thrift://[127.0.0.1:9083](http://127.0.0.1:9083/)");` for the dla new framework.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 26 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|8|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateCatalogStmt.java|0|18|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateCatalogStmt.java [hunk 1]
```diff
@@ -1,7 +1,63 @@
-// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
-
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/CreateCatalogStmt.java:2]: license

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateCatalogStmt.java [hunk 2]
```diff
@@ -1,7 +1,63 @@
...
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public String getCatalogType() {
+        return catalogType;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/CreateCatalogStmt.java:41]: we should use new analyzer. delete the old analyze

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -49,6 +49,7 @@ statement
     | SHOW DATABASES ((LIKE pattern=string) | (WHERE expression))?                          #showDatabases
     | GRANT identifierOrString TO user                                                      #grantRole
     | REVOKE identifierOrString FROM user                                                   #revokeRole
+    | CREATE EXTERNAL CATALOG identifierOrString properties                                 #createExternalCatalog
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:52]: change define position and add some comment about catalog

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 15 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|8|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateCatalogStmt.java|0|7|00.00%|

**Comment by DorianZheng** [comment]
Plz add ut

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 4]
```diff
@@ -122,6 +125,13 @@ alterSystemStatement
     : ALTER SYSTEM alterClause
     ;
 
+// ------------------------------------------- Catalog Statement -------------------------------------------------------
+
+createExternalCatalogStatement
+    :CREATE EXTERNAL CATALOG catalogName=identifierOrString properties
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:131]: add space

**Review by stephen-shelby — DISMISSED** [review]

**Review by DorianZheng — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]
LGTM

**Comment by stephen-shelby** [comment]
Need to complete the relevant implementation and then append ut

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 5]
```diff
@@ -122,6 +125,13 @@ alterSystemStatement
     : ALTER SYSTEM alterClause
     ;
 
+// ------------------------------------------- Catalog Statement -------------------------------------------------------
+
+createExternalCatalogStatement
+    : CREATE EXTERNAL CATALOG catalogName=identifierOrString properties
+    ;
+
+
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:134]: delete another \n

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 6]
```diff
@@ -1047,6 +1048,19 @@ public ParseNode visitRevokeRole(StarRocksParser.RevokeRoleContext context) {
...
+        Identifier identifier = (Identifier) visit(context.identifierOrString());
+        Map<String, String> properties = new HashMap<>();
+        if (context.properties() != null) {
+            List<Property> propertyList = visit(context.properties().property(), Property.class);
+            for (Property property : propertyList) {
+                properties.put(property.getKey(), property.getValue());
+            }
+        }
+        return new CreateCatalogStmt(identifier.getValue(), properties);
+    }
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1062]: code position should be before 341 line . Consistent with StarRocks.g4

**Review by dirtysalt — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Comment by caneGuy** [comment]
run starrocks_be_unittest

**Comment by caneGuy** [comment]
run starrocks_clang-format

**Comment by caneGuy** [comment]
run starrocks_clang-tidy

**Comment by caneGuy** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 16 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateCatalogStmt.java|0|7|00.00%|

**Pull request merged**